pkg/api/unversioned.go

```diff
@@ -22,16 +22,49 @@ import (
 
 // This file contains API types that are unversioned.
 
-// APIVersions lists the api versions that are available, to allow
-// version negotiation. APIVersions isn't just an unnamed array of
-// strings in order to allow for future evolution, though unversioned
+// APIVersions lists the versions that are available, to allow
+// clients to discover the API. This is returned at /api
 type APIVersions struct {
+	// Versions are the api versions that are available.
 	Versions []string `json:"versions"`
 }
 
+type APIGroupList struct {
+	Groups []APIGroup `json:"groups"`
+}
+
+type APIGroup struct {
+	GroupName        string    `json:"groupName"`
+	Versions         []Version `json:"versions"`
```

Comment:
Alternatively pod groupVersion string inside version struct, and client can
scan it to find their group version (secondary index)

On Sep 16, 2015, at 8:23 PM, Chao Xu <notifications@github.com> wrote:

In pkg/api/unversioned.go
<https://github.com/kubernetes/kubernetes/pull/13955#discussion_r39700982>:

>  	Versions []string `json:"versions"`
>  }
>
> +type APIGroupList struct {
> +	Groups []APIGroup `json:"groups"`
> +}
> +
> +type APIGroup struct {
> +	GroupName        string    `json:"groupName"`
> +	Versions         []Version `json:"versions"`

Also, following this logic, should we also make GroupVersion a struct
rather than an array of string?

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/13955/files#r39700982>.